### PR TITLE
Added arm-gcc-10-2020-q4 revision

### PR DESCRIPTION
**Pull request by ValentiWorkLearning**
The new stable version of gnu-arm toolchain is available. Because of some C++ features and bugfixes that are supplied with 10.2 gcc version it would be fine to update the existing GitHub action for adding the support of the new compiler.

### CODE REVIEW

**Comment by gretel** [comment]
@ValentiWorkLearning thank you!

**Comment by ValentiWorkLearning** [comment]
@gretel I've no idea, how to fix the issue with NodeJS download, because of passed build for Mac OS. How to reproduce these failed tests locally? I want to fix them and prepare this PR to be merged, because some essential fixes are supplied only from GCC 10.2( Especially  some of upcoming stable C++20 features ).

**Comment by gretel** [comment]
> @gretel I've no idea, how to fix the issue with NodeJS download, because of passed build for Mac OS.

not sure if i get you right. this error happens sometimes because `github actions` is using `docker`. there is nothing you can do to change that. but there is a workaround which i have added on my branch (`Take the trash out` step). please check the action running fine on my fork.

> How to reproduce these failed tests locally? I want to fix them and prepare this PR to be merged, because some essential fixes are supplied only from GCC 10.2( Especially some of upcoming stable C++20 features ).

one step at a time. we did port all what is required to have this working inlcuding on `osx` now.
it is not any different for the platforms when they actually use the same compilers. i also don't think any more abstraction (like build containers) should be put in place cause this would just add more hidden complexity and requires additional maintenance efforts.

the test suite completes [fine](https://github.com/gretel/EmuFlight/runs/1583787488?check_suite_focus=true). now, the tests have to test the right things.

nobody has taken any care in regards to unit testing since.. the original fork. all the changes, all features, the dozens of little experiements eventually got merged without working on the tests. the betaflight project is not accepting any PRs without the tests to complete fine - and this project would be better of doing same.

you just can't fix that on your own in a couple of days. the contributors of this project have to accept the need for unit testing and continue to work on their submissions. or, as some prefer - abandon their previous hackings and work on betaflight instead.

**Comment by gretel** [comment]
@ValentiWorkLearning thanks, there is nothing you can do. the post gets done automatically on reference - it's just how github works.

but, i was mixing up context cause what you wrote als matched the actual issue i was trying to comment on 😕

**Comment by gretel** [comment]
@fiam howdy! can you check on this please?

**Comment by ValentiWorkLearning** [comment]
@gretel seems these issue will be here for a long. If you need a quick temporary fix- these can be used:
```
    - name: Get the latest arm-gcc-none-eabi-gcc
      uses: wei/wget@v1
      id: download-arm-gcc
      with:
        args: 'https://developer.arm.com/-/media/Files/downloads/gnu-rm/10-2020q4/gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2'

    - name: Unpack arm-gcc-compiler
      run: |
        mkdir toolchain
        tar -xjf gcc-arm-none-eabi-10-2020-q4-major-x86_64-linux.tar.bz2 --directory ${{ github.workspace }}/toolchain
```

It's not a cross-platform approach, but having the latest compiler is more important, imho. 
P.S. I used this for my project because arm-gcc-none-eabi is used only under linux environment.

**Comment by gretel** [comment]
@ValentiWorkLearning many thanks.

>  It's not a cross-platform approach

this is something i can change 😸

File: dist/index.js [hunk 1]
```diff
@@ -27157,1459 +27158,1459 @@ module.exports = !fMapEnabled ? () => 'w'
...
-    module.exports = bigInt;
-}
-
-//amd check
-if (typeof define === "function" && define.amd) {
-    define( function () {
-        return bigInt;
-    });
-}
+var bigInt = (function (undefined) {
```
Comment (dlech) [inline dist/index.js:27161]: this looks like a bunch of unnecessary whitespace changes

**Comment by dlech** [comment]
I think Windows if failing because the new Windows .zip file now has a top level `gcc-arm-none-eabi-10-2020-q4-major` directory like the Mac and Linux .bz files. It didn't have the before.

The tar extractor has an option to strip this top level `extractor = tar.x({strip: 1, C: directory})` but the unzipper doesn't have this option `extractor = unzipper.Extract({path: directory})` See https://github.com/ZJONSSON/node-unzipper/issues/109

**Comment by dlech** [comment]
So maybe the solution is to not `await fs.promises.mkdir(directory, {recursive: true})` except for on older windows .zip files and remove the strip option from the tar unzipper. Then after the files are extracted, rename the directory to the shorter name.

**Comment by ryanwinter** [comment]
I resolved this in a fork by checking the presence of the bin directory in the zip to decide how to set the path: https://github.com/fiam/arm-none-eabi-gcc/compare/master...ryanwinter:master

**Comment by ValentiWorkLearning** [comment]
@dlech thanks! How to properly integrate your fixes into the existing PR for passing the checks?

**Comment by dlech** [comment]
> How to properly integrate your fixes into the existing PR for passing the checks?

I think @ryanwinter did something like this already in their fork.

**Comment by ValentiWorkLearning** [comment]
Seems the issue is fixed now in master branch by the action maintainer.
